### PR TITLE
feat: Add 'Delete Vehicle' option to admin menu

### PR DESCRIPTION
**Pull request by LiamDormon**
This turns the vehicle spawner into a multiple choice giving the option to delete as well as spawn.

### CODE REVIEW

File: locale/en.json [hunk 1]
```diff
@@ -104,6 +107,11 @@
                 "dialog_info": "Trying to spawn %{modelName}",
                 "onesync_error": "This option requires that OneSync be on for server side spawning"
             },
+            "delete_veh": {
+                "list_secondary": "Delete your vehicle",
+                "dialog_success": "Vehicle Deleted!",
+                "dialog_error": "You are not currently in a vehicle!"
```
Comment (Poggicek) [inline locale/en.json:113]: ~~"You are currently not in a vehicle!" sounds better. I might be wrong, correct me if so. I'm not a native english speaker~~ Nvm. I've said to myself too many times and now it loses it's meaning to me. Looking at stats, "not currently" is used way more

File: menu/src/state/vehiclemode.state.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  SPAWN = "spawn",
+  DELETE = "delete",
+}
+
+const vehicleMode = atom({
+  key: "vehicleModeState",
+  default: VehicleMode.SPAWN,
+});
+
+export const useVehicleMode = () => useRecoilState(vehicleMode);
```
Comment (TasoOneAsia) [inline menu/src/state/vehiclemode.state.ts:13]: We don't really need this to be a remembered state probably. Fine to just be reset on mount

File: menu/src/components/MainPage/MainPageList.tsx [hunk 3]
```diff
@@ -225,6 +227,23 @@ export const MainPageList: React.FC = () => {
...
+          }
+        );
+      }
+
+      enqueueSnackbar(t("nui_menu.page_main.delete_veh.dialog_success"), {
+        variant: "info",
+      });
+    });
+  }
+
```
Comment (TasoOneAsia) [inline menu/src/components/MainPage/MainPageList.tsx]: We need to have a oneSync ctx check here as we are using server side game state awareness

**Review by TasoOneAsia — CHANGES_REQUESTED** [review]

File: menu/src/state/vehiclemode.state.ts (hunk 2 above)
Comment (LiamDormon) [reply menu/src/state/vehiclemode.state.ts:13]: This is how other multi-choice options are mananged, healmode, teleportmode and playermode so I kept that the same for this.

Comment (TasoOneAsia) [reply menu/src/state/vehiclemode.state.ts:13]: Yea we don't need that for those either, I implemented it with player mode as that is a persistent mode for players, the other ones are somewhat unnecessary.

File: menu/src/components/MainPage/MainPageList.tsx (hunk 3 above)
Comment (LiamDormon) [reply menu/src/components/MainPage/MainPageList.tsx]: Pushed a commit for this

File: menu/src/components/MainPage/MainPageList.tsx [hunk 4]
```diff
@@ -228,6 +228,11 @@ export const MainPageList: React.FC = () => {
   };
 
   const handleDeleteVehicle = () => {
+    // If onesync is disabled, show an error due to server side entity handling
+    if (!serverCtx.oneSync.status) {
+      return enqueueSnackbar(t('nui_menu.page_main.delete_veh.onesync_error'), { variant: 'error' })
```
Comment (TasoOneAsia) [inline menu/src/components/MainPage/MainPageList.tsx]: Lets just have a combined locale key for both this and spawn vehicle as they are practically the same.

**Review by TasoOneAsia — CHANGES_REQUESTED** [review]

File: menu/src/components/MainPage/MainPageList.tsx (hunk 4 above)
Comment (LiamDormon) [reply menu/src/components/MainPage/MainPageList.tsx]: Pushed a commit combining them into one locale.

**Comment by TasoOneAsia** [comment]
Rebase against `feat/menu` and squash commits and this should be good, pending @tabarra review as well.

**Review by TasoOneAsia — APPROVED** [review]

**Pull request closed without merging**